src/Glpi/Asset/CustomFieldDefinition.php

```diff
@@ -158,10 +158,10 @@ private function validateSystemName(array &$input): bool
         global $DB;
 
         // Spaces are replaced with underscores and the name is made lowercase. Only lowercase letters and underscores are kept.
-        $input['name'] = preg_replace('/[^a-z_]/', '', strtolower(str_replace(' ', '_', $input['name'])));
+        $input['system_name'] = preg_replace('/[^a-z_]/', '', strtolower(str_replace(' ', '_', $input['system_name'])));
         // The name cannot start with an underscore
-        $input['name'] = ltrim($input['name'], '_');
-        if ($input['name'] === '') {
+        $input['system_name'] = ltrim($input['system_name'], '_');
```

Comment:
IMHO, as we are already doing these replacement on client side, we should probably just validate the pattern here, the same way it is done in the `AbstractDefinition::prepareInput()` method.

People who will try to script the creation of custom fields will probably prefer to have an error than an uncontrolled replacement.